projects/dsp-ui/src/lib/viewer/values/time-value/time-input/time-input.component.ts

```diff
@@ -19,6 +21,21 @@ import { CalendarDate, CalendarPeriod, GregorianCalendarDate } from 'jdnconverti
 import { Subject } from 'rxjs';
 import { CustomRegex } from '../../custom-regex';
 
+/** A valid time value must have both a date and a time, or both inputs must be null */
+export function dateTimeValidator(otherControl: FormControl): ValidatorFn {
+    return (control: AbstractControl): { [key: string]: any } | null => {
+
+        // valid if both date and time are null or have values
+        let invalid = !(control.value === null && otherControl.value === null || control.value !== null && otherControl.value !== null);
+
+        // ensure the time value does not contain an empty string
+        if (control.value === "" || otherControl.value === "") {
```

Comment:
Couldn't this be written all in one expression?

Reply:
done in a210b88